src/Compilers/CSharp/Test/Semantic/SourceGeneration/SyntaxAwareGeneratorTests.cs

```diff
@@ -1298,19 +1298,19 @@ class C
 
             var testGenerator = new PipelineCallbackGenerator(context =>
             {
-                var source = context.Sources.Syntax.Transform(c => c is FieldDeclarationSyntax fds, c =>
+                var source = context.SyntaxProvider.CreateSyntaxProvider((c, _) => c is FieldDeclarationSyntax fds, (c, _) =>
```

Comment:
>SyntaxProvider.CreateSyntaxProvider [](http://example.com/codeflow?start=37&length=35)

It seems surprising to have a `SyntaxProvider` with a `CreateSyntaxProvider()` instance method.

Reply:
This was the decision of the API review. 🤷